src/project/internal/notationproject.cpp

```diff
@@ -566,16 +567,17 @@ mu::Ret NotationProject::doSave(const io::path_t& path, engraving::MscIoMode ioM
 
     // Step 1: check writable
     {
-        if (fileSystem()->exists(savePath) && !fileSystem()->isWritable(savePath)) {
+        if ((fileSystem()->exists(savePath) && !fileSystem()->isWritable(savePath))
+            || (fileSystem()->exists(targetContainerPath) && !fileSystem()->isWritable(targetContainerPath))) {
             LOGE() << "failed save, not writable path: " << savePath;
-            return make_ret(notation::Err::UnknownError);
+            return make_ret(io::Err::FSWriteError);
         }
 
         if (ioMode == engraving::MscIoMode::Dir) {
             // Dir needs to be created, otherwise we can't move to it
             if (!QDir(targetContainerPath).mkpath(".")) {
                 LOGE() << "Couldn't create container directory";
```

Comment:
here better add the filename `LOGE() << "Couldn't create container directory: " << targetContainerPath;`

Reply:
Thanks again! Added this a new commit.